lib/srv/discovery/discovery.go

```diff
@@ -680,6 +688,16 @@ func (s *Server) initGCPWatchers(ctx context.Context, matchers []types.GCPMatche
 	return nil
 }
 
+// existingEICENodes returns all the EICE Nodes
+func (s *Server) existingEICENodes() []types.Server {
+	return s.nodeWatcher.GetNodes(s.ctx, func(n services.Node) bool {
+		labels := n.GetAllLabels()
+		_, accountOK := labels[types.AWSAccountIDLabel]
+		_, instanceOK := labels[types.AWSInstanceIDLabel]
+		return accountOK && instanceOK && n.GetSubKind() == types.SubKindOpenSSHEICENode
```

Comment:
Just wondering, why is it not enough to check for just subkind?

Reply:
If the labels don't exist we won't be able to compare against the received EC2 instances.
Is it possible to have an EICE node without those labels? It shouldn't, but I would rather be defensive here.